src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.managed.cs

```diff
@@ -514,7 +514,7 @@ void UnfillBackward()
 				while (firstMaterializedLine != null && GetMeasuredEnd(firstMaterializedLine.FirstView) < ExtendedViewportStart + extentAdjustment)
 				{
 					// Dematerialize lines that are entirely outside extended viewport
-					RecycleLine(firstMaterializedLine, clearContainer: false);
+					RecycleLine(firstMaterializedLine, clearContainer: true);
```

Comment:
I would have guess that other tests would have failed as a result, but that does not seem like it. Could you try and reproduce something for this? If containers are cleared, we should be getting cleared datacontext changes for generated containers on a large scroll.

Reply:
Indeed, there was a UI test failing because of this. I've converted the UI test to a runtime test (and kept the UI test because it also took screenshots and stuff) and moved the additional cleanup (of visual states and indices) to the lighter and always triggered PrepareForRecycle rather than the heavier and conditionally triggered `CleanUpContainerOverride` which also clears the DataContext etc. This way, we don't need to make any additional calls with `cleanContainer: true` and we still get the most necessary cleanups each time.